pkg/server/builder/resource/types.go

```diff
@@ -88,10 +90,10 @@ type StatusSubResource interface {
 	CopyTo(parent ObjectWithStatusSubResource)
 }
 
-// ArbitrarySubResource defines interface for registering arbitrary subresource to the parent resource.
-type ArbitrarySubResource interface {
+// GenericSubResource defines interface for registering arbitrary subresource to the parent resource.
+type GenericSubResource interface {
 	Name() string
-	// TODO: fill the details for this interface.
+	GetStorageProvider(parentObj Object, rootPath string, fs filesystem.FS, parentWatchSet *filesystem.WatchSet, parentSP apiserver.StorageProvider) apiserver.StorageProvider
```

Comment:
To allow for custom storage implementation via GetStorageProvider method, I moved and made public a few types from the filepath package to filesystem package. 

I initially started with an implementation that was using the FS and WatchSet types, but ended up relying on parent storage pretty much exclusively, so I am happy to revert that part if necessary.  I decided to leave it in this PR because I thought other sub-resource types might need access to these interfaces more than mine.

Reply:
hmmmm....how is your implementation using `fs` and `parentWatchSet`?

The Kubernetes apiserver APIs for resources are usually more high-level than this, and don't usually require specific details about how the underlying objects are stored... but maybe i'm missing something here...

Reply:
I agree. When I looked at the PR just after I submitted it, I thought it probably makes more sense to solely rely on the parent storage. Let me rework this PR and verify that my subresources still work, but I think they will. This will also eliminate the need to change the filepath/filesystem types.

Reply:
Done. The change is much simpler/smaller now and still allows me to do everything I need for DCP.